### ISSUE TITLE
Not using latest revision

### ISSUE THREAD

**Issue by digitalghost-dev** [google-github-actions/deploy-cloudrun#479]
### TL;DR

Out of the box, this action used to be able to deploy the latest revision but now, it's not anymore. I set the optional flag of `no_traffic` to `false` and it's still not deploying the latest revision when I look in the console.

### Expected behavior

Deploy the latest revision.

### Observed behavior

The latest revision is **not** being deployed.

### Action YAML

```yaml
- name: Deploy Image
        id: 'deploy'
        uses: 'google-github-actions/deploy-cloudrun@v1'
        with: 
          service: streamlit
          image: "${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ secrets.PROJECT_ID }}/${{ secrets.REGISTRY_REPO }}/streamlit:${{ env.VERSION_NUMBER }}"
          flags: "--service-account=${{ secrets.WIF_SERVICE_ACCOUNT_CLOUD_RUN }} --max-instances=25 --cpu-boost"
          no_traffic: false
```

### Log output

```
2023-12-13T14:58:45.0305453Z Requested labels: ubuntu-22.04
2023-12-13T14:58:45.0305706Z Job defined at: digitalghost-dev/premier-league/.github/workflows/ci_streamlit.yaml@refs/heads/main
2023-12-13T14:58:45.0305834Z Waiting for a runner to pick up this job...
2023-12-13T14:58:45.7570162Z Job is waiting for a hosted runner to come online.
2023-12-13T14:58:48.4685219Z Job is about to start running on the hosted runner: GitHub Actions 2 (hosted)
2023-12-13T14:58:50.6197285Z Current runner version: '2.311.0'
2023-12-13T14:58:50.6290334Z ##[group]Operating System
2023-12-13T14:58:50.6290986Z Ubuntu
2023-12-13T14:58:50.6291466Z 22.04.3
2023-12-13T14:58:50.6291809Z LTS
2023-12-13T14:58:50.6292245Z ##[endgroup]
2023-12-13T14:58:50.6292652Z ##[group]Runner Image
2023-12-13T14:58:50.6293108Z Image: ubuntu-22.04
2023-12-13T14:58:50.6293608Z Version: 20231211.1.0
2023-12-13T14:58:50.6294621Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20231211.1/images/ubuntu/Ubuntu2204-Readme.md
2023-12-13T14:58:50.6296131Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20231211.1
2023-12-13T14:58:50.6297076Z ##[endgroup]
2023-12-13T14:58:50.6297491Z ##[group]Runner Image Provisioner
2023-12-13T14:58:50.6298083Z 2.0.321.1
2023-12-13T14:58:50.6298439Z ##[endgroup]
2023-12-13T14:58:50.6299331Z ##[group]GITHUB_TOKEN Permissions
2023-12-13T14:58:50.6300914Z Contents: read
2023-12-13T14:58:50.6301377Z Metadata: read
2023-12-13T14:58:50.6302084Z ##[endgroup]
2023-12-13T14:58:50.6305428Z Secret source: Actions
2023-12-13T14:58:50.6306144Z Prepare workflow directory
2023-12-13T14:58:50.7038858Z Prepare all required actions
2023-12-13T14:58:50.7194667Z Getting action download info
2023-12-13T14:58:50.9235401Z Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
2023-12-13T14:58:52.0990187Z Download action repository 'google-github-actions/auth@v1' (SHA:3a3c4c57d294ef65efaaee4ff17b22fa88dd3c69)
2023-12-13T14:58:52.4924629Z Download action repository 'google-github-actions/deploy-cloudrun@v2' (SHA:82a0d44f48151e2d393811779576ec6ad902146b)
2023-12-13T14:58:52.9626746Z Complete job name: deploy-streamlit-image
2023-12-13T14:58:53.0503759Z ##[group]Run actions/checkout@v4
2023-12-13T14:58:53.0504253Z with:
2023-12-13T14:58:53.0504901Z   repository: digitalghost-dev/***
2023-12-13T14:58:53.0505471Z   token: ***
2023-12-13T14:58:53.0505906Z   ssh-strict: true
2023-12-13T14:58:53.0506304Z   persist-credentials: true
2023-12-13T14:58:53.0506706Z   clean: true
2023-12-13T14:58:53.0507140Z   sparse-checkout-cone-mode: true
2023-12-13T14:58:53.0507554Z   fetch-depth: 1
2023-12-13T14:58:53.0507890Z   fetch-tags: false
2023-12-13T14:58:53.0508337Z   show-progress: true
2023-12-13T14:58:53.0508695Z   lfs: false
2023-12-13T14:58:53.0509095Z   submodules: false
2023-12-13T14:58:53.0509474Z   set-safe-directory: true
2023-12-13T14:58:53.0509839Z env:
2023-12-13T14:58:53.0510241Z   GAR_LOCATION: us-central1
2023-12-13T14:58:53.0510631Z   VERSION_NUMBER: 2.12.1
2023-12-13T14:58:53.0511121Z   REGISTRY_IMAGE: digitalghostdev/***
2023-12-13T14:58:53.0511617Z ##[endgroup]
2023-12-13T14:58:55.5471430Z Syncing repository: digitalghost-dev/***
2023-12-13T14:58:55.5473237Z ##[group]Getting Git version info
2023-12-13T14:58:55.5474152Z Working directory is '/home/runner/work/***/***'
2023-12-13T14:58:55.5474934Z [command]/usr/bin/git version
2023-12-13T14:58:55.7466240Z git version 2.43.0
2023-12-13T14:58:55.7493197Z ##[endgroup]
2023-12-13T14:58:55.7583422Z Temporarily overriding HOME='/home/runner/work/_temp/3e83a425-5f05-4cdf-b11e-13d53b0ea890' before making global git config changes
2023-12-13T14:58:55.7585319Z Adding repository directory to the temporary git global config as a safe directory
2023-12-13T14:58:55.7591519Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/***/***
2023-12-13T14:58:55.7634270Z Deleting the contents of '/home/runner/work/***/***'
2023-12-13T14:58:55.7639661Z ##[group]Initializing the repository
2023-12-13T14:58:55.7644011Z [command]/usr/bin/git init /home/runner/work/***/***
2023-12-13T14:58:55.8241192Z hint: Using 'master' as the name for the initial branch. This default branch name
2023-12-13T14:58:55.8242808Z hint: is subject to change. To configure the initial branch name to use in all
2023-12-13T14:58:55.8244283Z hint: of your new repositories, which will suppress this warning, call:
2023-12-13T14:58:55.8245386Z hint: 
2023-12-13T14:58:55.8246467Z hint: 	git config --global init.defaultBranch <name>
2023-12-13T14:58:55.8247994Z hint: 
2023-12-13T14:58:55.8249167Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2023-12-13T14:58:55.8251286Z hint: 'development'. The just-created branch can be renamed via this command:
2023-12-13T14:58:55.8253102Z hint: 
2023-12-13T14:58:55.8253763Z hint: 	git branch -m <name>
2023-12-13T14:58:55.8323428Z Initialized empty Git repository in /home/runner/work/***/***/.git/
2023-12-13T14:58:55.8333999Z [command]/usr/bin/git remote add origin https://github.com/digitalghost-dev/***
2023-12-13T14:58:55.8442410Z ##[endgroup]
2023-12-13T14:58:55.8443292Z ##[group]Disabling automatic garbage collection
2023-12-13T14:58:55.8446604Z [command]/usr/bin/git config --local gc.auto 0
2023-12-13T14:58:55.8475799Z ##[endgroup]
2023-12-13T14:58:55.8476396Z ##[group]Setting up auth
2023-12-13T14:58:55.8481022Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-12-13T14:58:55.8509633Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-12-13T14:58:56.0800997Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-12-13T14:58:56.0833730Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-12-13T14:58:56.1072472Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2023-12-13T14:58:56.1108733Z ##[endgroup]
2023-12-13T14:58:56.1110204Z ##[group]Fetching the repository
2023-12-13T14:58:56.1118394Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +b27764a2a24fec3912aba0a8625483cb17d31fb0:refs/remotes/origin/main
2023-12-13T14:58:56.5794948Z From https://github.com/digitalghost-dev/***
2023-12-13T14:58:56.5796931Z  * [new ref]         b27764a2a24fec3912aba0a8625483cb17d31fb0 -> origin/main
2023-12-13T14:58:56.5916999Z ##[endgroup]
2023-12-13T14:58:56.5917942Z ##[group]Determining the checkout info
2023-12-13T14:58:56.5919167Z ##[endgroup]
2023-12-13T14:58:56.5920040Z ##[group]Checking out the ref
2023-12-13T14:58:56.5925416Z [command]/usr/bin/git checkout --progress --force -B main refs/remotes/origin/main
2023-12-13T14:58:56.6076556Z Switched to a new branch 'main'
2023-12-13T14:58:56.6077977Z branch 'main' set up to track 'origin/main'.
2023-12-13T14:58:56.6084596Z ##[endgroup]
2023-12-13T14:58:56.6119296Z [command]/usr/bin/git log -1 --format='%H'
2023-12-13T14:58:56.6142951Z 'b27764a2a24fec3912aba0a8625483cb17d31fb0'
2023-12-13T14:58:56.6449622Z ##[group]Run google-github-actions/auth@v1
2023-12-13T14:58:56.6450144Z with:
2023-12-13T14:58:56.6450573Z   token_format: access_token
2023-12-13T14:58:56.6451736Z   workload_identity_provider: ***
2023-12-13T14:58:56.6452533Z   service_account: ***
2023-12-13T14:58:56.6452950Z   project_id: ***
2023-12-13T14:58:56.6453339Z   create_credentials_file: true
2023-12-13T14:58:56.6453835Z   export_environment_variables: true
2023-12-13T14:58:56.6454276Z   cleanup_credentials: true
2023-12-13T14:58:56.6454698Z   access_token_lifetime: 3600s
2023-12-13T14:58:56.6455350Z   access_token_scopes: https://www.googleapis.com/auth/cloud-platform
2023-12-13T14:58:56.6456139Z   retries: 3
2023-12-13T14:58:56.6456714Z   backoff: 250
2023-12-13T14:58:56.6457080Z   id_token_include_email: false
2023-12-13T14:58:56.6457489Z env:
2023-12-13T14:58:56.6457868Z   GAR_LOCATION: us-central1
2023-12-13T14:58:56.6458289Z   VERSION_NUMBER: 2.12.1
2023-12-13T14:58:56.6458876Z   REGISTRY_IMAGE: digitalghostdev/***
2023-12-13T14:58:56.6459326Z ##[endgroup]
2023-12-13T14:58:58.9260752Z Created credentials file at "/home/runner/work/***/***/gha-creds-547e49bc79f910ac.json"
2023-12-13T14:58:59.2550645Z ##[group]Run google-github-actions/deploy-cloudrun@v2
2023-12-13T14:58:59.2551400Z with:
2023-12-13T14:58:59.2551815Z   service: streamlit
2023-12-13T14:58:59.2552588Z   image: us-central1-docker.pkg.dev/***/***/streamlit:2.12.1
2023-12-13T14:58:59.2553549Z   flags: --service-account=*** --max-instances=25 --cpu-boost
2023-12-13T14:58:59.2554158Z   no_traffic: false
2023-12-13T14:58:59.2554518Z   region: us-central1
2023-12-13T14:58:59.2554977Z   skip_default_labels: false
2023-12-13T14:58:59.2555369Z env:
2023-12-13T14:58:59.2555693Z   GAR_LOCATION: us-central1
2023-12-13T14:58:59.2556172Z   VERSION_NUMBER: 2.12.1
2023-12-13T14:58:59.2556663Z   REGISTRY_IMAGE: digitalghostdev/***
2023-12-13T14:58:59.2557559Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/***/***/gha-creds-547e49bc79f910ac.json
2023-12-13T14:58:59.2558596Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/***/***/gha-creds-547e49bc79f910ac.json
2023-12-13T14:58:59.2559716Z   GOOGLE_GHA_CREDS_PATH: /home/runner/work/***/***/gha-creds-547e49bc79f910ac.json
2023-12-13T14:58:59.2560396Z   CLOUDSDK_CORE_PROJECT: ***
2023-12-13T14:58:59.2560873Z   CLOUDSDK_PROJECT: ***
2023-12-13T14:58:59.2561375Z   GCLOUD_PROJECT: ***
2023-12-13T14:58:59.2561798Z   GCP_PROJECT: ***
2023-12-13T14:58:59.2562565Z   GOOGLE_CLOUD_PROJECT: ***
2023-12-13T14:58:59.2562945Z ##[endgroup]
2023-12-13T14:59:00.9970348Z [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/b9f06dc8-e8d4-493d-afc5-2f150ce93dc4 -f /home/runner/work/_temp/09c66dc0-bdc0-46e1-a179-0b1507080e21
2023-12-13T14:59:15.4123791Z Successfully authenticated
2023-12-13T14:59:15.4127332Z Running: gcloud run deploy streamlit --quiet --image us-central1-docker.pkg.dev/***/***/streamlit:2.12.1 --update-labels managed-by=github-actions,commit-sha=b27764a2a24fec3912aba0a8625483cb17d31fb0 --platform managed --format json --region us-central1 --service-account *** --max-instances 25 --cpu-boost
2023-12-13T14:59:19.4744649Z Post job cleanup.
2023-12-13T14:59:19.5678013Z Removed exported credentials at "/home/runner/work/***/***/gha-creds-547e49bc79f910ac.json".
2023-12-13T14:59:19.6302100Z Post job cleanup.
2023-12-13T14:59:19.7068882Z [command]/usr/bin/git version
2023-12-13T14:59:19.7108542Z git version 2.43.0
2023-12-13T14:59:19.7151440Z Temporarily overriding HOME='/home/runner/work/_temp/519c5730-9774-4200-af6d-a01501edbd85' before making global git config changes
2023-12-13T14:59:19.7153557Z Adding repository directory to the temporary git global config as a safe directory
2023-12-13T14:59:19.7157809Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/***/***
2023-12-13T14:59:19.7191782Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-12-13T14:59:19.7223902Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-12-13T14:59:19.7521447Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-12-13T14:59:19.7542316Z http.https://github.com/.extraheader
2023-12-13T14:59:19.7553757Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2023-12-13T14:59:19.7584787Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-12-13T14:59:19.8038834Z Cleaning up orphan processes

```

**Comment by sethvargo** [comment]
Hi @digitalghost-dev 

Please provide the debug logs and ensure you're using the latest version (`@v2`).

**Comment by digitalghost-dev** [comment]
@sethvargo - Okay, I updated the action to use `v2` but still doing the same. I updated my original post with the logs from the output. I don't have debug logging enabled on my workflow but if you need those, I can try and set that up.

**Comment by sethvargo** [comment]
Hi @digitalghost-dev - we need the [debug logs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) to assist further.

**Comment by digitalghost-dev** [comment]
@sethvargo - I've added them. Thanks for linking the documentation.

**Comment by sethvargo** [comment]
The action is running:

```shell
gcloud run deploy streamlit \
  --quiet \
  --image us-central1-docker.pkg.dev/***/***/streamlit:2.12.1 \
  --update-labels managed-by=github-actions,commit-sha=b27764a2a24fec3912aba0a8625483cb17d31fb0 \
  --platform managed \
  --format json \
  --region us-central1 \
  --service-account *** \
  --max-instances 25 \
  --cpu-boost
```

Are you expecting the action to do something different?

**Comment by digitalghost-dev** [comment]
I just wanted it to deploy the new revision that is deployed by the action. I could've sworn that this happened automatically a few months ago but by looking at the `README.md`, it looks like I need to use the `gcloud run services update-traffic` command. Something like this:
```bash
gcloud run services update-traffic streamlit --to-latest
```

Is this correct? Do I use the `revision_traffic` input with this action for that?

**Comment by sethvargo** [comment]
Is it not deploying `us-central1-docker.pkg.dev/***/***/streamlit:2.12.1`?

**Comment by digitalghost-dev** [comment]
It's deploying it but just not setting it to the latest revision:
![image](https://github.com/google-github-actions/deploy-cloudrun/assets/86637723/70c432e1-0f6e-4327-be13-f2092a842e93)

**Comment by sethvargo** [comment]
> After a deployment with this flag the LATEST revision will not receive
        traffic on future deployments. To restore sending traffic to the LATEST
        revision by default, run the gcloud run services update-traffic command
        with --to-latest.

It sounds like you may have changed the default configuration?

**Comment by digitalghost-dev** [comment]
The default configuration with the workflow `yaml` file? I can look at my history for that file and see if something sticks out as to what I could've changed. 

If that's not case, how do I run that command with this action?

```
gcloud run services update-traffic streamlit --to-latest
```

**Comment by sethvargo** [comment]
Hi @digitalghost-dev - you can use `setup-gcloud` and run the command. You should only need to run that command once to make Cloud Run always use the latest revision moving forward (based on my understanding of the documentation).

**Comment by digitalghost-dev** [comment]
Gotcha okay, thanks for the recommendation. I'll give that action a go and run that command after deployment.

**Comment by sethvargo** [comment]
Sounds good - please re-open if it's still occurring. If so, that would indicate an issue with Cloud Run and I can point you to their support queue.

**Comment by digitalghost-dev** [comment]
Hi, @sethvargo - Just to give you an update, so I ran the `gcloud run services update-traffic streamlit --to-latest` commanded on my local machine and I checked the Cloud Run console after and noticed that there is a `(to latest)` flag now by the latest revision. I then ran the normal CI workflow I have and it deployed the latest revision like you said, just need to run that command once and it's set.

I must've deleted the service and created it again without running that command hence the original issue. Thanks again for your help.
